crates/chain-state/src/memory_overlay.rs

```diff
@@ -17,14 +18,12 @@ use std::collections::HashMap;
 /// the historical state provider for fallback lookups.
 #[allow(missing_debug_implementations)]
 pub struct MemoryOverlayStateProvider {
-    /// The collection of executed parent blocks. Expected order is newest to oldest.
-    pub(crate) in_memory: Vec<ExecutedBlock>,
-    /// The collection of hashed state from in-memory blocks.
-    pub(crate) hashed_post_state: HashedPostState,
-    /// The collection of aggregated in-memory trie updates.
-    pub(crate) trie_updates: TrieUpdates,
     /// Historical state provider for state lookups that are not found in in-memory blocks.
     pub(crate) historical: Box<dyn StateProvider>,
+    /// The collection of executed parent blocks. Expected order is newest to oldest.
+    pub(crate) in_memory: Vec<ExecutedBlock>,
+    /// Lazy-loaded in-memory trie data.
+    pub(crate) trie_state: Mutex<Option<MemoryOverlayTrieState>>,
```

Comment:
this type isn't shared, right?

so would a OnceLock be sufficient here?

Reply:
nvm @DaniPopes was again faster -.-

Reply:
already fixed 👍